src/devices/Adg731/Metadata/Adg731Metadata.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Device.Spi;
+
+namespace Iot.Device.Adg731
+{
+    /// <summary>
+    /// Metadata class for ADG731
+    /// </summary>
+    public class Adg731Metadata : IDeviceMetadata, ISpiDeviceMetadata, IMultiplexerDeviceMetadata
+    {
+        /// <inheritdoc/>
+        public string Manufacturer => "Analog Devices";
+
+        /// <inheritdoc/>
+        public string Product => "ADG731";
+
+        /// <inheritdoc/>
+        public string ProductCategory => "Analog Multiplexer";
+
+        /// <inheritdoc/>
+        public string ProductDescription => "32-Channel, Serially Controlled 4 Ohm 1.8 V to 5.5 V, +/- 2.5 V, Analog Multiplexer";
```

Comment:
IMO:
- datasheet link should be in comment and README.md
- ValidSpiModes should be an IEnumerable and be directly on the ADC (as static method: i.e. GetValidSpiModes which can return cached array or something) - reason for IEnumerable is because SpiMode is not `[Flags]` enum
-